Test fail if no timezone is set on system

**Describe the bug**

Running the tests on a system with no timezone set makes `pyhanko_tests/cli_tests/conftest.py fail`:

```
================================================================== ERRORS ==================================================================
______________________________________________________ ERROR collecting test session _______________________________________________________
/usr/lib/python3.10/importlib/__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
<frozen importlib._bootstrap>:1050: in _gcd_import
    ???
<frozen importlib._bootstrap>:1027: in _find_and_load
    ???
<frozen importlib._bootstrap>:1006: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:688: in _load_unlocked
    ???
/usr/lib/python3.10/site-packages/_pytest/assertion/rewrite.py:172: in exec_module
    exec(co, module.__dict__)
pyhanko_tests/cli_tests/conftest.py:52: in <module>
    FREEZE_DT = tzlocal.get_localzone().localize(datetime.datetime(2020, 8, 1))
E   AttributeError: 'datetime.timezone' object has no attribute 'localize'
========================================================= short test summary info ==========================================================
ERROR  - AttributeError: 'datetime.timezone' object has no attribute 'localize'
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
============================================================= 1 error in 1.45s =============================================================
```

This is due to the following deprecated use of datetime:

```python
>>> import datetime
>>> import tzlocal
>>> tzlocal.get_localzone().localize(datetime.datetime(2020, 8, 1))
/usr/lib/python3.10/site-packages/tzlocal/unix.py:183: UserWarning: Can not find any timezone configuration, defaulting to UTC.
  warnings.warn("Can not find any timezone configuration, defaulting to UTC.")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'datetime.timezone' object has no attribute 'localize'
```

On a system with a timezone set, this part still produces a deprecation warning:

```python
>>> import datetime
>>> import tzlocal
>>> FREEZE_DT = tzlocal.get_localzone().localize(datetime.datetime(2020, 8, 1))
<stdin>:1: PytzUsageWarning: The localize method is no longer necessary, as this time zone supports the fold attribute (PEP 495). For more details on migrating to a PEP 495-compliant implementation, see https://pytz-deprecation-shim.readthedocs.io/en/latest/migration.html
```

See this for more info: https://pytz-deprecation-shim.readthedocs.io/en/latest/migration.html

**To Reproduce**

Running in a docker image is the easiest way to reproduce.

**Expected behavior**

Tests should pass, regardless of if a timezone is set or not.

**Environment (please complete the following information):**
 - OS: Arch Linux
 - Version N/A

--- a/pyhanko/pdf_utils/misc.py
+++ b/pyhanko/pdf_utils/misc.py
@@ -10,12 +10,12 @@ classes.
 
 import os
 from dataclasses import dataclass
-from datetime import datetime
+from datetime import datetime, timezone
 from enum import Enum
 from io import BytesIO
 from typing import Callable, Generator, Iterable, Optional, TypeVar, Union
 
-from pyhanko_certvalidator.util import ConsList
+from pyhanko_certvalidator.util import CancelableAsyncIterator, ConsList
 
 __all__ = [
     'PdfError',
@@ -49,9 +49,6 @@ __all__ = [
     'isoparse',
     'lift_iterable_async',
 ]
-
-import pytz
-from pyhanko_certvalidator import CancelableAsyncIterator
 
 DEFAULT_CHUNK_SIZE = 4096
 """
@@ -542,7 +539,7 @@ def isoparse(dt_str: str) -> datetime:
     dt = parse(dt_str)
     if dt.tzinfo is None:
         # assume UTC
-        dt = dt.replace(tzinfo=pytz.UTC)
+        dt = dt.replace(tzinfo=timezone.utc)
     return dt
 
 

--- a/pyhanko_tests/cli_tests/conftest.py
+++ b/pyhanko_tests/cli_tests/conftest.py
@@ -1,7 +1,6 @@
 import datetime
 import logging
 import os
-import warnings
 
 import pytest
 import requests_mock
@@ -47,9 +46,7 @@ CERTOMANCER_ARCHITECTURES = {
     "ed25519": TESTING_CA_ED25519,
     "ed448": TESTING_CA_ED448,
 }
-with warnings.catch_warnings():
-    warnings.simplefilter("ignore")
-    FREEZE_DT = tzlocal.get_localzone().localize(datetime.datetime(2020, 8, 1))
+FREEZE_DT = datetime.datetime(2020, 8, 1, tzinfo=tzlocal.get_localzone())
 
 
 @pytest.fixture(scope="module", params=list(CERTOMANCER_ARCHITECTURES))

--- a/pyhanko_tests/test_cms.py
+++ b/pyhanko_tests/test_cms.py
@@ -1,12 +1,11 @@
 import hashlib
 import itertools
 import os
-from datetime import datetime
+from datetime import datetime, timezone
 from io import BytesIO
 from typing import Optional
 
 import pytest
-import pytz
 import tzlocal
 import yaml
 from asn1crypto import algos, cms, core, keys
@@ -276,7 +275,7 @@ async def test_detached_cms_with_tst():
     assert status.signer_reported_dt is None
     assert status.timestamp_validity.intact
     assert status.timestamp_validity.valid
-    assert status.timestamp_validity.timestamp == datetime.now(tz=pytz.utc)
+    assert status.timestamp_validity.timestamp == datetime.now(tz=timezone.utc)
     assert 'The TSA certificate is untrusted' in status.pretty_print_details()
     assert status.valid
     assert status.intact
@@ -302,12 +301,12 @@ async def test_detached_cms_with_content_tst():
     assert status.signer_reported_dt is None
     assert status.timestamp_validity.intact
     assert status.timestamp_validity.valid
-    assert status.timestamp_validity.timestamp == datetime.now(tz=pytz.utc)
+    assert status.timestamp_validity.timestamp == datetime.now(tz=timezone.utc)
     assert status.content_timestamp_validity
     assert status.content_timestamp_validity.intact
     assert status.content_timestamp_validity.valid
     assert status.content_timestamp_validity.timestamp == datetime.now(
-        tz=pytz.utc
+        tz=timezone.utc
     )
     pretty_print = status.pretty_print_details()
     assert 'The TSA certificate is untrusted' in pretty_print
@@ -359,12 +358,12 @@ async def test_detached_cms_with_wrong_tst():
     assert not status.timestamp_validity.intact
     assert status.timestamp_validity.valid
 
-    assert status.timestamp_validity.timestamp == datetime.now(tz=pytz.utc)
+    assert status.timestamp_validity.timestamp == datetime.now(tz=timezone.utc)
     assert status.content_timestamp_validity
     assert status.content_timestamp_validity.intact
     assert status.content_timestamp_validity.valid
     assert status.content_timestamp_validity.timestamp == datetime.now(
-        tz=pytz.utc
+        tz=timezone.utc
     )
     assert status.valid
     assert status.intact
@@ -405,13 +404,13 @@ async def test_detached_cms_with_wrong_content_tst():
     assert status.signer_reported_dt is None
     assert status.timestamp_validity.intact
     assert status.timestamp_validity.valid
-    assert status.timestamp_validity.timestamp == datetime.now(tz=pytz.utc)
+    assert status.timestamp_validity.timestamp == datetime.now(tz=timezone.utc)
     assert status.content_timestamp_validity
     assert not status.content_timestamp_validity.intact
     # internally, the content timestamp is OK
     assert status.content_timestamp_validity.valid
     assert status.content_timestamp_validity.timestamp == datetime.now(
-        tz=pytz.utc
+        tz=timezone.utc
     )
     pretty_print = status.pretty_print_details()
     assert 'The TSA certificate is untrusted' in pretty_print
@@ -1725,10 +1724,10 @@ async def test_detached_cades_cms_with_tst():
     status = await async_validate_detached_cms(
         b'Hello world!', signature['content']
     )
-    assert status.signer_reported_dt == datetime.now(tz=pytz.utc)
+    assert status.signer_reported_dt == datetime.now(tz=timezone.utc)
     assert status.timestamp_validity.intact
     assert status.timestamp_validity.valid
-    assert status.timestamp_validity.timestamp == datetime.now(tz=pytz.utc)
+    assert status.timestamp_validity.timestamp == datetime.now(tz=timezone.utc)
     assert 'The TSA certificate is untrusted' in status.pretty_print_details()
     assert status.valid
     assert status.intact

--- a/pyhanko_tests/test_diff_analysis.py
+++ b/pyhanko_tests/test_diff_analysis.py
@@ -1,10 +1,9 @@
 import os
-from datetime import datetime
+from datetime import datetime, timezone
 from io import BytesIO
 from itertools import product
 
 import pytest
-import pytz
 from freezegun.api import freeze_time
 
 from pyhanko.pdf_utils import generic, misc
@@ -74,7 +73,7 @@ def test_no_changes_policy():
 
     w = IncrementalPdfFileWriter(out)
     # do an /Info update
-    dt = generic.pdf_date(datetime(2020, 10, 10, tzinfo=pytz.utc))
+    dt = generic.pdf_date(datetime(2020, 10, 10, tzinfo=timezone.utc))
     info = generic.DictionaryObject({pdf_name('/CreationDate'): dt})
     w.set_info(info)
     w.write_in_place()
@@ -116,7 +115,7 @@ def test_double_sig_add_field(file_ix):
     # create a new signature field after signing
     w = IncrementalPdfFileWriter(out)
     # throw in an /Info update for good measure
-    dt = generic.pdf_date(datetime(2020, 10, 10, tzinfo=pytz.utc))
+    dt = generic.pdf_date(datetime(2020, 10, 10, tzinfo=timezone.utc))
     info = generic.DictionaryObject({pdf_name('/CreationDate'): dt})
     w.set_info(info)
 
@@ -163,7 +162,7 @@ def test_double_sig_different_pages():
 
     w = IncrementalPdfFileWriter(out)
     # throw in an /Info update for good measure
-    dt = generic.pdf_date(datetime(2020, 10, 10, tzinfo=pytz.utc))
+    dt = generic.pdf_date(datetime(2020, 10, 10, tzinfo=timezone.utc))
     info = generic.DictionaryObject({pdf_name('/CreationDate'): dt})
     w.set_info(info)
     w.write_in_place()

--- a/pyhanko_tests/test_meta.py
+++ b/pyhanko_tests/test_meta.py
@@ -1,10 +1,9 @@
 import itertools
 import os
-from datetime import datetime
+from datetime import datetime, timezone
 from io import BytesIO
 
 import pytest
-import pytz
 import tzlocal
 from freezegun import freeze_time
 
@@ -16,6 +15,12 @@ from pyhanko.pdf_utils.misc import StringWithLanguage
 from pyhanko.pdf_utils.reader import PdfFileReader
 from pyhanko.pdf_utils.writer import PdfFileWriter, copy_into_new_writer
 from pyhanko_tests.samples import MINIMAL, PDF_DATA_DIR, VECTOR_IMAGE_PDF
+
+try:
+    import zoneinfo
+except ImportError:
+    # noinspection PyUnresolvedReferences
+    from backports import zoneinfo
 
 
 def test_no_meta_view_fails_gracefully():
@@ -73,7 +78,9 @@ def test_incremental_update_meta_view():
 @freeze_time(datetime(2022, 9, 7, tzinfo=tzlocal.get_localzone()))
 @pytest.mark.parametrize('writer_type', ('fresh', 'from_data', 'incremental'))
 def test_writer_meta_view_does_not_persist_changes(writer_type):
-    exp_value = datetime(2020, 9, 5, 19, 30, 57, tzinfo=pytz.FixedOffset(120))
+    exp_value = datetime(
+        2020, 9, 5, 19, 30, 57, tzinfo=zoneinfo.ZoneInfo('CET')
+    )
     if writer_type == 'fresh':
         w = PdfFileWriter()
         exp_value = None
@@ -96,7 +103,7 @@ def test_writer_meta_view_does_not_persist_changes(writer_type):
                 {'title': 'Test test'},
                 {'author': 'John Doe'},
                 {'keywords': ['foo', 'bar', 'baz']},
-                {'created': datetime(2022, 9, 7, tzinfo=pytz.utc)},
+                {'created': datetime(2022, 9, 7, tzinfo=timezone.utc)},
                 {'author': 'John Doe', 'subject': 'Blah blah blah'},
             ],
             ("pdf1.7", "pdf2.0"),

--- a/pyhanko_tests/test_pades.py
+++ b/pyhanko_tests/test_pades.py
@@ -1,10 +1,9 @@
 import asyncio
-from datetime import datetime
+from datetime import datetime, timezone
 from io import BytesIO
 from typing import Iterable
 
 import pytest
-import pytz
 from asn1crypto import cms, core, tsp
 from certomancer.integrations.illusionist import Illusionist
 from certomancer.registry import ArchLabel, CertLabel, KeyLabel
@@ -837,7 +836,7 @@ def test_add_revinfo_and_timestamp(requests_mock):
             {'trust_roots': TRUST_ROOTS},
         )
         assert status.valid and status.trusted
-        assert status.signer_reported_dt == datetime.now(tz=pytz.utc)
+        assert status.signer_reported_dt == datetime.now(tz=timezone.utc)
 
         # ... but PAdES-LTA should fail
         with pytest.raises(
@@ -884,7 +883,7 @@ def test_add_revinfo_and_lta_timestamp(requests_mock):
             {'trust_roots': TRUST_ROOTS},
         )
         assert status.valid and status.trusted
-        assert status.signer_reported_dt == datetime.now(tz=pytz.utc)
+        assert status.signer_reported_dt == datetime.now(tz=timezone.utc)
 
     # test post-expiration, but before timestamp expires
     with freeze_time('2025-11-01'):

--- a/pyhanko_tests/test_utils.py
+++ b/pyhanko_tests/test_utils.py
@@ -1,13 +1,13 @@
 import datetime
 import itertools
 import os
+from datetime import timezone
 from fractions import Fraction
 from io import BytesIO
 from itertools import product
 from typing import Tuple
 
 import pytest
-import pytz
 
 import pyhanko.pdf_utils.extensions
 from pyhanko.pdf_utils import generic, misc, writer
@@ -24,6 +24,12 @@ from pyhanko.pdf_utils.reader import PdfFileReader, RawPdfPath
 from pyhanko.pdf_utils.rw_common import PdfHandler
 
 from .samples import *
+
+try:
+    import zoneinfo
+except ImportError:
+    # noinspection PyUnresolvedReferences
+    from backports import zoneinfo
 
 
 @pytest.mark.parametrize(
@@ -490,7 +496,7 @@ def test_box_constraint_recalc():
 def test_trailer_update():
     w = IncrementalPdfFileWriter(BytesIO(MINIMAL_ONE_FIELD))
     w._update_meta = lambda: None
-    dt = generic.pdf_date(datetime.datetime(2020, 10, 10, tzinfo=pytz.utc))
+    dt = generic.pdf_date(datetime.datetime(2020, 10, 10, tzinfo=timezone.utc))
 
     info = generic.DictionaryObject({pdf_name('/CreationDate'): dt})
     w.trailer['/Info'] = w.add_object(info)
@@ -576,7 +582,7 @@ TESTDATE_CET = datetime.datetime(
     hour=1,
     minute=5,
     second=59,
-    tzinfo=pytz.timezone('CET'),
+    tzinfo=zoneinfo.ZoneInfo('Europe/Brussels'),
 )
 
 TESTDATE_EST = datetime.datetime(
@@ -586,7 +592,7 @@ TESTDATE_EST = datetime.datetime(
     hour=1,
     minute=5,
     second=59,
-    tzinfo=pytz.timezone('EST'),
+    tzinfo=zoneinfo.ZoneInfo('America/New_York'),
 )
 
 
@@ -617,7 +623,7 @@ TESTDATE_EST = datetime.datetime(
                 hour=1,
                 minute=5,
                 second=59,
-                tzinfo=pytz.utc,
+                tzinfo=timezone.utc,
             ),
         ),
         ('D:20080203010559+01\'00', TESTDATE_CET),
@@ -1401,17 +1407,29 @@ def test_parse_name_invalid_utf8():
     [
         (
             datetime.datetime(
-                2020, 12, 26, 15, 5, 11, tzinfo=pytz.timezone('EST')
+                2020,
+                12,
+                26,
+                15,
+                5,
+                11,
+                tzinfo=zoneinfo.ZoneInfo('America/New_York'),
             ),
             "D:20201226150511-05'00'",
         ),
         (
-            datetime.datetime(2020, 12, 26, 15, 5, 11, tzinfo=pytz.utc),
+            datetime.datetime(2020, 12, 26, 15, 5, 11, tzinfo=timezone.utc),
             "D:20201226150511Z",
         ),
         (
             datetime.datetime(
-                2020, 12, 26, 15, 5, 11, tzinfo=pytz.timezone('CET')
+                2020,
+                12,
+                26,
+                15,
+                5,
+                11,
+                tzinfo=zoneinfo.ZoneInfo('Europe/Brussels'),
             ),
             "D:20201226150511+01'00'",
         ),

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -35,9 +35,8 @@ classifiers = [
 requires-python = ">=3.7"
 dependencies = [
     "asn1crypto>=1.5.1",
-    "pytz>=2020.1",
     "qrcode>=6.1",
-    "tzlocal>=2.1",
+    "tzlocal>=4.3",
     "pyhanko-certvalidator==0.22.*",
     "click>=7.1.2",
     "requests>=2.24.0",
@@ -82,6 +81,7 @@ testing-basic = [
     "pytest-asyncio==0.21.0",
     "certomancer~=0.9.1",
     "pytest-cov~=4.0.0",
+    "backports.zoneinfo[tzdata];python_version<\"3.9\"",
 ]
 testing = [
     "pyHanko[testing-basic,pkcs11,xmp,opentype,image-support,async-http,extra-pubkey-algs]",
@@ -100,7 +100,6 @@ mypy = [
     "pyHanko[pkcs11,xmp,opentype,image-support,async-http,extra-pubkey-algs]",
     "types-requests",
     "types-python-dateutil",
-    "types-pytz",
     "types-PyYAML",
     "types-tzlocal",
 ]

